client/asset/dcr/dcr.go

```diff
@@ -5119,28 +5126,142 @@ func (dcr *ExchangeWallet) SetVSP(url string) error {
 
 // PurchaseTickets purchases n number of tickets. Part of the asset.TicketBuyer
 // interface.
-func (dcr *ExchangeWallet) PurchaseTickets(n int, feeSuggestion uint64) ([]*asset.Ticket, error) {
+func (dcr *ExchangeWallet) PurchaseTickets(n int, feeSuggestion uint64) error {
 	if n < 1 {
-		return nil, nil
+		return nil
 	}
 	if !dcr.connected.Load() {
-		return nil, errors.New("not connected, login first")
+		return errors.New("not connected, login first")
 	}
 	// I think we need to set this, otherwise we probably end up with default
 	// of DefaultRelayFeePerKb = 1e4 => 10 atoms/byte.
 	feePerKB := dcrutil.Amount(dcr.feeRateWithFallback(feeSuggestion) * 1000)
 	if err := dcr.wallet.SetTxFee(dcr.ctx, feePerKB); err != nil {
-		return nil, fmt.Errorf("error setting wallet tx fee: %w", err)
+		return fmt.Errorf("error setting wallet tx fee: %w", err)
+	}
+
```

Comment:
If you check check for insufficient balance over here and return an error the UI could show insufficient balance instead of a check mark first and then immediately fail after.

Reply:
We can't really calculate exact fees, but we can apply a simpler filter. On it.